config/option.go

```diff
@@ -0,0 +1,45 @@
+package config
+
+import (
+	"github.com/taskcluster/taskcluster-cli/extpoints"
+)
+
+// A ConfigOption is something with a default value and a validator.
+// Only requirement is that values are JSON structures.
+type ConfigOption struct {
```

Comment:
nit: golint recommends renaming it to something like `Option` because users will be accessing it as `config.ConfigOption`, which stutters.